gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/BloodPressureMapperTest.java

```diff
@@ -0,0 +1,201 @@
+
+        var bloodPressure = bloodPressureMapper.mapBloodPressure(ehrExtract, patient, ENCOUNTER_LIST).get(0);
+
+        assertFixedValues(bloodPressure);
+        assertThat(bloodPressure.getId()).isEqualTo(EXAMPLE_ID);
+        assertThat(bloodPressure.getCode().getCodingFirstRep().getDisplay()).isEqualTo(CODING_DISPLAY_MOCK);
+        assertThat(bloodPressure.hasSubject()).isTrue();
+        assertThat(bloodPressure.getIssuedElement().asStringValue()).isEqualTo(ISSUED_EXAMPLE);
+        assertThat(bloodPressure.getEffective().toString()).isEqualTo(DateFormatUtil.parseToDateTimeType(EFFECTIVE_EXAMPLE).toString());
+        assertThat(bloodPressure.getPerformer().get(0).getReference()).isEqualTo(PPRF_PARTICIPANT_ID);
+        assertThat(bloodPressure.getComment()).isEqualTo(COMMENT_EXAMPLE_1);
+
+        assertThat(bloodPressure.getComponent().get(0).getCode().getCodingFirstRep().getDisplay()).isEqualTo(CODING_DISPLAY_MOCK);
+        assertThat(bloodPressure.getComponent().get(0).getValueQuantity().getValue().toString())
+            .isEqualTo(COMPONENT_1_VALUE_QUANTITY_VALUE);
+        assertThat(bloodPressure.getComponent().get(0).getInterpretation().getText()).isEqualTo(COMPONENT_1_INTERPRETATION_TEXT);
+        assertThat(bloodPressure.getComponent().get(0).getReferenceRange().get(0).getText())
+            .isEqualTo(COMPONENT_1_REFERENCE_RANGE_TEXT);
+
+        assertThat(bloodPressure.getComponent().get(1).getCode().getCodingFirstRep().getDisplay()).isEqualTo(CODING_DISPLAY_MOCK);
+        assertThat(bloodPressure.getComponent().get(1).getValueQuantity().getValue().toString())
+            .isEqualTo(COMPONENT_2_VALUE_QUANTITY_VALUE);
+        assertThat(bloodPressure.getComponent().get(1).getInterpretation().getText()).isEqualTo(COMPONENT_2_INTERPRETATION_TEXT);
+        assertThat(bloodPressure.getComponent().get(1).getReferenceRange().get(0).getText()).isEqualTo(COMPONENT_2_REFERENCE_RANGE_TEXT);
+    }
+
+    @Test
+    public void mapBloodPressureWithNoOptionalData() {
+        when(codeableConceptMapper.mapToCodeableConcept(any())).thenReturn(CODEABLE_CONCEPT);
+
+        var ehrExtract = unmarshallEhrExtractElement("no_optional_data_bp_example.xml");
+
+        var bloodPressure = bloodPressureMapper.mapBloodPressure(ehrExtract, patient, ENCOUNTER_LIST).get(0);
+
+        assertFixedValues(bloodPressure);
+        assertThat(bloodPressure.getId()).isEqualTo(EXAMPLE_ID);
+        assertThat(bloodPressure.getCode().getCodingFirstRep().getDisplay()).isEqualTo(CODING_DISPLAY_MOCK);
+        assertThat(bloodPressure.hasSubject()).isTrue();
+        assertThat(bloodPressure.getIssuedElement().asStringValue()).isEqualTo(ISSUED_EXAMPLE);
+        assertThat(bloodPressure.getPerformer().get(0).getReference()).isEqualTo(PPRF_PARTICIPANT_ID);
+
+        assertThat(bloodPressure.getEffective()).isNull();
+        assertThat(StringUtils.isEmpty(bloodPressure.getComment()));
```

Comment:
This `assertThat` statement doesn't verify anything yet, you need to do either `     assertThat(StringUtils.isEmpty(bloodPressure.getComment())).isTrue();` or `        assertTrue(StringUtils.isEmpty(bloodPressure.getComment()));`. The same for lines 131, 132, 136, 137, 187, 198.